tests/ReactiveVarsCache.test.ts

```diff
@@ -0,0 +1,81 @@
+import { ReactiveVar } from "@apollo/client";
+import { cachedReactiveVarTypename, makeCachedVar } from "../src/cache/ReactiveVarsCache";
+import InvalidationPolicyCache from "../src/cache/InvalidationPolicyCache";
+
+describe('ReactiveVarsCache', () => {
+  let cache: InvalidationPolicyCache;
+  let rv: ReactiveVar<any>;
+
+  beforeEach(() => {
+    cache = new InvalidationPolicyCache();
+    rv = makeCachedVar<boolean>('test', false);
+  });
+
+  test('should cache value changes', () => {
+    expect(cache.extract(true, false)).toEqual({
+      "CachedReactiveVar:test": {
+        __typename: cachedReactiveVarTypename,
+        id: 'test',
+        value: false,
+      },
+      __META: {
+        extraRootIds: ['CachedReactiveVar:test']
+      }
+    });
+    rv(true);
+    expect(cache.extract(true, false)).toEqual({
+      "CachedReactiveVar:test": {
+        __typename: cachedReactiveVarTypename,
+        id: 'test',
+        value: true,
+      },
+      __META: {
+        extraRootIds: ['CachedReactiveVar:test']
+      }
+    });
+    rv(false);
+    expect(cache.extract(true, false)).toEqual({
+      "CachedReactiveVar:test": {
+        __typename: cachedReactiveVarTypename,
+        id: 'test',
+        value: false,
+      },
+      __META: {
+        extraRootIds: ['CachedReactiveVar:test']
+      }
+    });
+  });
+
+  test('should be initialized with the existing cached value', () => {
+    rv(true);
+    const rv2 = makeCachedVar('test', false);
```

Comment:
So as long as the supplied `id` matches one that exists in the cache, the existing cached value will be used even if a new value is supplied? (e.g. `false` is supplied here but the cached RV returns `true`)

Wouldn't this lead to some develop confusion when they unintentionally reuse a key that already exists?

Reply:
Yea the ids should be unique, here I was just using the same ID for testing purposes. If we're worried folks would pass in the same one as somewhere else in the app accidentally then it could throw an error or something. Naming it after the variable name could be a convention, open to ideas

Reply:
Agree that this could cause developer confusion.  Would it make sense to log an error if `makeCachedVar()` is invoked with an existing ID?


Do you think it's the responsibility of the library or the developer to guard against duplicate IDs?

If the former,